src/runtime/test/hydrate-no-encapsulation.spec.tsx

```diff
@@ -166,6 +166,69 @@ describe('hydrate no encapsulation', () => {
     `);
   });
 
+  it('nested text slot with key', async () => {
```

Comment:
this test is a reproduction of the issue I was seeing, if you wanted to see what happens you could check out this branch locally and either run the test without rebuilding or

```sh
git checkout origin/main -- src/runtime/vdom/vdom-render.ts
npm run build
npx jest --clearCache
npx --node-options=--experimental-vm-modules jest --coverage=false -- hydrate-no-encapsulation
```

which will run this test without the fix so you can see what happens

Reply:
I ran this locally, adding what I found for posterity:
```
  ● hydrate no encapsulation › nested text slot with key

    expect(received).toBe(expected) // Object.is equality

    - Expected  - 3
    + Received  + 1

      <cmp-a class="hydrated">
        <!--r.1-->
        <cmp-b class="hydrated">
    -     <!--r.2-->
    -     <!---->
    -     <!--s.2.0.0.0.-->
    +     <!---->
          light-dom
          <footer></footer>
        </cmp-b>
      </cmp-a>
```

Reply:
Yup that's what I expect to see 👍